boa_engine/src/builtins/intl/mod.rs

```diff
@@ -151,6 +151,41 @@ fn best_available_locale(available_locales: &[JsString], locale: &JsString) -> O
     }
 }
 
+/// Returns the position of the first found unicode locale extension in a given string.
+///
+/// If no extensions found, return the length of requested locale
+fn get_leftmost_unicode_extension_pos(requested_locale: &str) -> usize {
+    let ext_sep = "-u-";
+    let src_locale = requested_locale.to_lowercase();
+    let pos = src_locale.find(ext_sep);
+    match pos {
+        Some(idx) => idx,
+        None => src_locale.len(),
+    }
+}
```

Comment:
`suggestion`: You should use `icu::locid::Locale` and get `LanguageIdentifier` out of it.

Reply:
Removed `get_leftmost_unicode_extension_pos` completely.